### ISSUE TITLE
Error page after updating to NC 27 if using GroupFolders with FullTextSearch

### ISSUE THREAD

**Issue by MeiKatz** [nextcloud/groupfolders#2438]
### Steps to reproduce
1. Update to NextCloud 27
2. Try to open app Files
3. it crashes

### Expected behaviour
Files app should open without an error

### Actual behaviour
Error page is shown

### Server configuration

**Operating system:** Alpine Linux?

**Web server:** Apache2

**Database:** PGSQL

**PHP version:** 8.1.20

**Nextcloud version:** 27

**Group folders version:** 15.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** update, from NC 26

**Where did you install Nextcloud from:** AIO

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
OC\\App\\AppManager::checkAppForUser(): Argument #2 ($user) must be of type ?OCP\\IUser, string given, called in /var/www/html/lib/private/App/AppManager.php on line 307
```
```json
[
  {
    "file":"/var/www/html/lib/private/App/AppManager.php",
    "line":307,
    "function":"checkAppForUser",
    "class":"OC\\App\\AppManager",
    "type":"->",
    "args":[
      "yes",
      "*** username ***"
    ]
  },
  {
    "file":"/var/www/html/custom_apps/files_fulltextsearch/lib/Service/GroupFoldersService.php",
    "line":103,
    "function":"isEnabledForUser",
    "class":"OC\\App\\AppManager",
    "type":"->",
    "args":[
      "groupfolders",
      "*** username ***"
    ]
  },
  {
    "function":"__construct",
    "class":"OCA\\Files_FullTextSearch\\Service\\GroupFoldersService",
    "type":"->",
    "args":[
      "*** username ***",
      [
        "OC\\DB\\ConnectionAdapter"
      ],
      [
        "OC\\App\\AppManager"
      ],
      [
        "OC\\Share20\\Manager"
      ],
      [
        "OC\\Group\\Manager"
      ],
      [
        "OCA\\Files_FullTextSearch\\Service\\LocalFilesService"
      ],
      [
        "OCA\\Files_FullTextSearch\\Service\\ConfigService",
        [
          "1",
          "0",
          "0",
          "0",
          "0",
          "20",
          "1",
          "1",
          "0",
          "0",
          2
        ]
      ],
      [
        "OCA\\Files_FullTextSearch\\Service\\MiscService"
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":116,
    "function":"newInstanceArgs",
    "class":"ReflectionClass",
    "type":"->",
    "args":[
      [
        "*** username ***",
        [
          "OC\\DB\\ConnectionAdapter"
        ],
        [
          "OC\\App\\AppManager"
        ],
        [
          "OC\\Share20\\Manager"
        ],
        [
          "OC\\Group\\Manager"
        ],
        [
          "OCA\\Files_FullTextSearch\\Service\\LocalFilesService"
        ],
        [
          "OCA\\Files_FullTextSearch\\Service\\ConfigService",
          [
            "1",
            "0",
            "0",
            "0",
            "0",
            "20",
            "1",
            "1",
            "0",
            "0",
            2
          ]
        ],
        [
          "OCA\\Files_FullTextSearch\\Service\\MiscService"
        ]
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":124,
    "function":"buildClass",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      [
        "ReflectionClass",
        "OCA\\Files_FullTextSearch\\Service\\GroupFoldersService"
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":142,
    "function":"resolve",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Service\\GroupFoldersService"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php",
    "line":494,
    "function":"query",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Service\\GroupFoldersService"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php",
    "line":466,
    "function":"queryNoFallback",
    "class":"OC\\AppFramework\\DependencyInjection\\DIContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Service\\GroupFoldersService"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":97,
    "function":"query",
    "class":"OC\\AppFramework\\DependencyInjection\\DIContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Service\\GroupFoldersService",
      true
    ]
  },
  {
    "function":"OC\\AppFramework\\Utility\\{closure}",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      "*** sensitive parameters replaced ***"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":116,
    "function":"array_map",
    "args":[
      [
        "Closure"
      ],
      [
        [
          "ReflectionParameter",
          "container"
        ],
        [
          "ReflectionParameter",
          "rootFolder"
        ],
        [
          "ReflectionParameter",
          "appManager"
        ],
        [
          "ReflectionParameter",
          "userManager"
        ],
        [
          "ReflectionParameter",
          "shareManager"
        ],
        [
          "ReflectionParameter",
          "urlGenerator"
        ],
        [
          "ReflectionParameter",
          "commentsManager"
        ],
        [
          "ReflectionParameter",
          "systemTagObjectMapper"
        ],
        [
          "ReflectionParameter",
          "systemTagManager"
        ],
        [
          "ReflectionParameter",
          "configService"
        ],
        [
          "ReflectionParameter",
          "localFilesService"
        ],
        [
          "ReflectionParameter",
          "externalFilesService"
        ],
        "*** sensitive parameters replaced ***",
        [
          "ReflectionParameter",
          "extensionService"
        ],
        [
          "ReflectionParameter",
          "fullTextSearchManager"
        ],
        [
          "ReflectionParameter",
          "miscService"
        ]
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":124,
    "function":"buildClass",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      [
        "ReflectionClass",
        "OCA\\Files_FullTextSearch\\Service\\FilesService"
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":142,
    "function":"resolve",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Service\\FilesService"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php",
    "line":494,
    "function":"query",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Service\\FilesService"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php",
    "line":466,
    "function":"queryNoFallback",
    "class":"OC\\AppFramework\\DependencyInjection\\DIContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Service\\FilesService"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":97,
    "function":"query",
    "class":"OC\\AppFramework\\DependencyInjection\\DIContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Service\\FilesService",
      true
    ]
  },
  {
    "function":"OC\\AppFramework\\Utility\\{closure}",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      "*** sensitive parameters replaced ***"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":116,
    "function":"array_map",
    "args":[
      [
        "Closure"
      ],
      [
        [
          "ReflectionParameter",
          "l10n"
        ],
        [
          "ReflectionParameter",
          "configService"
        ],
        "*** sensitive parameters replaced ***",
        [
          "ReflectionParameter",
          "searchService"
        ],
        [
          "ReflectionParameter",
          "extensionService"
        ],
        [
          "ReflectionParameter",
          "miscService"
        ]
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":124,
    "function":"buildClass",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      [
        "ReflectionClass",
        "OCA\\Files_FullTextSearch\\Provider\\FilesProvider"
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php",
    "line":142,
    "function":"resolve",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Provider\\FilesProvider"
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php",
    "line":494,
    "function":"query",
    "class":"OC\\AppFramework\\Utility\\SimpleContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Provider\\FilesProvider"
    ]
  },
  {
    "file":"/var/www/html/lib/private/ServerContainer.php",
    "line":155,
    "function":"queryNoFallback",
    "class":"OC\\AppFramework\\DependencyInjection\\DIContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Provider\\FilesProvider"
    ]
  },
  {
    "file":"/var/www/html/custom_apps/fulltextsearch/lib/Service/ProviderService.php",
    "line":122,
    "function":"query",
    "class":"OC\\ServerContainer",
    "type":"->",
    "args":[
      "OCA\\Files_FullTextSearch\\Provider\\FilesProvider"
    ]
  },
  {
    "file":"/var/www/html/custom_apps/fulltextsearch/lib/Service/ProviderService.php",
    "line":311,
    "function":"loadProvider",
    "class":"OCA\\FullTextSearch\\Service\\ProviderService",
    "type":"->",
    "args":[
      "files_fulltextsearch",
      "OCA\\Files_FullTextSearch\\Provider\\FilesProvider"
    ]
  },
  {
    "file":"/var/www/html/custom_apps/fulltextsearch/lib/Service/ProviderService.php",
    "line":282,
    "function":"loadProvidersFromList",
    "class":"OCA\\FullTextSearch\\Service\\ProviderService",
    "type":"->",
    "args":[
      "files_fulltextsearch",
      [
        "OCA\\Files_FullTextSearch\\Provider\\FilesProvider"
      ]
    ]
  },
  {
    "file":"/var/www/html/custom_apps/fulltextsearch/lib/Service/ProviderService.php",
    "line":102,
    "function":"loadProvidersFromApp",
    "class":"OCA\\FullTextSearch\\Service\\ProviderService",
    "type":"->",
    "args":[
      "files_fulltextsearch"
    ]
  },
  {
    "file":"/var/www/html/custom_apps/fulltextsearch/lib/Service/ProviderService.php",
    "line":156,
    "function":"loadProviders",
    "class":"OCA\\FullTextSearch\\Service\\ProviderService",
    "type":"->",
    "args":[
      
    ]
  },
  {
    "file":"/var/www/html/custom_apps/fulltextsearch/lib/Capabilities.php",
    "line":80,
    "function":"getConfiguredProviders",
    "class":"OCA\\FullTextSearch\\Service\\ProviderService",
    "type":"->",
    "args":[
      
    ]
  },
  {
    "file":"/var/www/html/lib/private/CapabilitiesManager.php",
    "line":74,
    "function":"getCapabilities",
    "class":"OCA\\FullTextSearch\\Capabilities",
    "type":"->",
    "args":[
      
    ]
  },
  {
    "file":"/var/www/html/lib/private/Template/JSConfigHelper.php",
    "line":166,
    "function":"getCapabilities",
    "class":"OC\\CapabilitiesManager",
    "type":"->",
    "args":[
      false,
      true
    ]
  },
  {
    "file":"/var/www/html/lib/private/TemplateLayout.php",
    "line":240,
    "function":"getConfig",
    "class":"OC\\Template\\JSConfigHelper",
    "type":"->",
    "args":[
      
    ]
  },
  {
    "file":"/var/www/html/lib/private/legacy/OC_Template.php",
    "line":182,
    "function":"__construct",
    "class":"OC\\TemplateLayout",
    "type":"->",
    "args":[
      "user",
      "files"
    ]
  },
  {
    "file":"/var/www/html/lib/public/AppFramework/Http/TemplateResponse.php",
    "line":213,
    "function":"fetchPage",
    "class":"OC_Template",
    "type":"->",
    "args":[
      [
        0,
        "*** username ***",
        "*** fullname ***",
        false,
        "yes",
        "basename",
        "asc",
        "0",
        0,
        1,
        0,
        [
          "OCP\\Template"
        ],
        [
          [
            "files",
            "*** truncated ***"
          ],
          [
            "recent",
            "*** truncated ***"
          ],
          [
            "favorites",
            "*** truncated ***"
          ],
          [
            "sharingout",
            "*** truncated ***"
          ],
          [
            "sharingin",
            "*** truncated ***"
          ],
          [
            "deletedshares",
            "*** truncated ***"
          ],
          [
            "pendingshares",
            "*** truncated ***"
          ],
          [
            "shareoverview",
            "*** truncated ***"
          ],
          [
            "systemtagsfilter",
            "*** truncated ***"
          ]
        ],
        [
          
        ]
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php",
    "line":182,
    "function":"render",
    "class":"OCP\\AppFramework\\Http\\TemplateResponse",
    "type":"->",
    "args":[
      
    ]
  },
  {
    "file":"/var/www/html/lib/private/AppFramework/App.php",
    "line":183,
    "function":"dispatch",
    "class":"OC\\AppFramework\\Http\\Dispatcher",
    "type":"->",
    "args":[
      [
        "OCA\\Files\\Controller\\ViewController"
      ],
      "index"
    ]
  },
  {
    "file":"/var/www/html/lib/private/Route/Router.php",
    "line":315,
    "function":"main",
    "class":"OC\\AppFramework\\App",
    "type":"::",
    "args":[
      "OCA\\Files\\Controller\\ViewController",
      "index",
      [
        "OC\\AppFramework\\DependencyInjection\\DIContainer"
      ],
      [
        "files.view.index"
      ]
    ]
  },
  {
    "file":"/var/www/html/lib/base.php",
    "line":1064,
    "function":"match",
    "class":"OC\\Route\\Router",
    "type":"->",
    "args":[
      "/apps/files/"
    ]
  },
  {
    "file":"/var/www/html/index.php",
    "line":36,
    "function":"handleRequest",
    "class":"OC",
    "type":"::",
    "args":[
      
    ]
  }
]
```
</details>

**Comment by vasyugan** [comment]
This is obviously an interoperability issue between fulltextsearch and groupfolders, see here  https://github.com/nextcloud/fulltextsearch/issues/754

**Comment by MeiKatz** [comment]
Thanks for your reply. I could disable the app for full text search and enabled the group folders app again until the conflict is solved.

**Comment by bastikus** [comment]
I don't have fulltextsearch app, but groupfolders app doesn't work in NC 27.
`OCP\AppFramework\QueryException: Could not resolve OCA\GroupFolders\Folder\FolderManager! Class "OCA\GroupFolders\Folder\FolderManager" does not exist`

**Comment by koelle25** [comment]
> I don't have fulltextsearch app, but groupfolders app doesn't work in NC 27. And it was fresh installation NC 27, not update. `OCP\AppFramework\QueryException: Could not resolve OCA\GroupFolders\Folder\FolderManager! Class "OCA\GroupFolders\Folder\FolderManager" does not exist`

@bastikus This seems to be a different issue (at least it's a different exception). I recommend opening a separate issue for this.

**Comment by bastikus** [comment]
@koelle25 Yes, you are right. Thank you! This issue https://github.com/nextcloud/groupfolders/issues/2478

**Comment by rainerjung** [comment]
Note that this is not fixed by #2476.

**Comment by joshtrichards** [comment]
Given this and following comments, is this issue now fixed for those on this thread?

https://github.com/nextcloud/fulltextsearch/issues/754#issuecomment-1680411407

If not, please post your exact:
- NC server version
- Groupfolders app version
- fulltextsearch app version
